### PR TITLE
New credentials

### PR DESCRIPTION
**Pull request by tonisives**
HMKit can be intialized with either `HMKitOAuthCredentials` or `HMKitPrivateKeyCredentials`.

`HMKitOAuthCredentials`:

```java
    HMKitCredentials credentials = new HMKitOAuthCredentials(
      "client_id",
      "client_secret"
    );

    HMKitConfiguration configuration = new HMKitConfiguration.Builder()
      .credentials(credentials)
      .environment(HMKitFleet.Environment.SANDBOX)
      .build();

    HMKitFleet hmkit = new HMKitFleet(configuration);
```

`HMKitPrivateKeyCredentials`

```java
    HMKitCredentials credentials = new HMKitPrivateKeyCredentials(
      "client_id",
      "private_key_json_file_contents"
    );

    HMKitConfiguration configuration = new HMKitConfiguration.Builder()
      .credentials(credentials)
      .environment(HMKitFleet.Environment.SANDBOX)
      .build();

    HMKitFleet hmkit = new HMKitFleet(configuration);
```

### CODE REVIEW

**Comment by slashmili** [comment]
@tonisives what do you think in case of private key credentials, we actually ask for private key and private key id?

```kotlin
    HMKitCredentials credentials = new HMKitPrivateKeyCredentials(
      "client_id",
      "private_key_id",
      "private_key_pem"
    );
```

**Comment by tonisives** [comment]
> @tonisives what do you think in case of private key credentials, we actually ask for private key and private key id?
> 
> ```kotlin
>     HMKitCredentials credentials = new HMKitPrivateKeyCredentials(
>       "client_id",
>       "private_key_id",
>       "private_key_pem"
>     );
> ```

It might be easier for users to copy/paste the json file contents.
We had a problem where users didn't include the `-----BEGIN PRIVATE KEY-----\n` part.

But negative is that if we change the json file keys/content then the parsing in hmkit-fleet fails

Either option is fine with me

**Comment by slashmili** [comment]
I understand your concern. However I'd say people make mistake and then learn(or read the doc) and use it properly :D 

Let's go for explicit configuration and only pass what is needed.

File: hmkit-fleet/build.gradle [hunk 1]
```diff
@@ -28,37 +28,32 @@ dependencies {
...
 
     if (!project.hasProperty('depLocation') || project.depLocation > 0) {
         api deps.crypto
     } else {
         api project(":hmkit-crypto-telematics")
     }
 
-    // web
-    api('com.squareup.okhttp3:okhttp:4.10.0')
+    api('com.squareup.okhttp3:okhttp:4.12.0')
```
Comment (slashmili) [inline hmkit-fleet/build.gradle:37]: do we still need Hmkit-crypto?
```
api project(":hmkit-crypto-telematics")
```

File: hmkit-fleet/src/main/kotlin/HMKitCredentials.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }.toString()
+  }
+}
+
+internal fun getJwt(
+  privateKey: String,
+  privateKeyId: String,
+  jwtProvider: HMKitCredentials.JwtProvider
+): String {
+  val crypto: Crypto = jwtProvider.getCrypto()
```
Comment (slashmili) [inline hmkit-fleet/src/main/kotlin/HMKitCredentials.kt]: Now that we are using vanilla  JWT without our crypto logic, I'd suggest to use one of the JWT libraries that is available in Kotlin or Java land. They are more tested and receive update regularly

**Review by slashmili — APPROVED** [review]
LGTM 🚀

**Pull request merged**